csrc/scheduler/transpose.cpp

```diff
@@ -475,8 +473,8 @@ std::string getTransposeRuntimeRejectReason(
     return "Transpose scheduler does not perform well on small problem sizes.";
   }
 
-  auto inner_size1 = shape_in_ref1[inner_most_pos1_in_ref1];
-  auto inner_size2 = shape_in_ref1[inner_most_pos2_in_ref1];
+  auto inner_size1 = shape_in_ref1.at(inner_most_pos1_in_ref1);
```

Comment:
IIUC, `shape_in_ref1` is based on the loop domain of the reference, whereas `inner_most_pos1_in_ref1` is based on its allocation domain. I think we are implicitly assuming the loop domain is a permutation of the allocation domain. If that's the case, I think it'd be safe to have an assertion.

Also, what if the innermost allocation ID is DID parallelized? That ID should be ignored for the sake of the analysis here, I believe.

Reply:
> whereas inner_most_pos1_in_ref1 is based on its allocation domain

I'm unsure about this. https://github.com/NVIDIA/Fuser/blob/2638286389ec0cea59d0d01469c13b9a21d8a66b/csrc/scheduler/transpose.cpp#L427 indicates inner_most_pos1_in_ref1 is supposed to be a loop axis.

Reply:
Ah, you're right